src/R/Support/Impl/Help/IntelliSenseRHost.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Languages.Editor.Shell;
+using Microsoft.R.Host.Client;
+using Microsoft.R.Support.Settings;
+
+namespace Microsoft.R.Support.Help {
+    [Export(typeof(IIntellisenseRHost))]
+    public sealed class IntelliSenseRHost : IIntellisenseRHost {
+        private static readonly Guid SessionId = new Guid("8BEF9C06-39DC-4A64-B7F3-0C68353362C9");
+        private readonly ICoreShell _coreShell;
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly SemaphoreSlim _sessionSemaphore = new SemaphoreSlim(1, 1);
+
+        [ImportingConstructor]
+        public IntelliSenseRHost(ICoreShell coreShell, IRSessionProvider sessionProvider) {
```

Comment:
To @AlexanderSher - If session stops it is probably not good since fetching of signatures and quick infos must be quick, it can't wait for another session to spin up.